src/registry/mod.rs

```diff
@@ -77,10 +72,10 @@ where T: Decodable, String: Borrow<K>, K: Hash + Eq + Display {
 }
 
 /// Set a key in the registry to a particular value
-pub fn set<T: ToJson>(key: String, flags: AccessFlags, val: T) {
-    trace!("set: {}", key);
-    let ref mut write_reg = *write_lock();
+pub fn set<T: ToTable>(key: String, flags: AccessFlags, val: T) {
+    trace!("set: {:?} {}", flags, key);
     let regvalue = RegistryValue::new(flags, val);
+    let ref mut write_reg = *write_lock();
```

Comment:
Why are you taking the reference of a dereferenced value? Just do:
`let mut write_reg = write_lock();`